ramp4-pcar4/ramp4-pcar4
# Global Layer Time Defaults

### Related Item(s)

#2248

### Changes

- Adds instance-level default configuration for layer warning times.
- Increased the default time for load warnings from 4 to 10 seconds
- Squashed a mild whoops in timer manager code
- Added a no-layer sample for QA / Load Testing

### Notes

Chose 10 as new default, since WFS & File sources do all the heavy lifting in the load phase.

### Testing

Steps:
1. Start with default Sample 1. Once you see RAMP sort-of appear, start counting seconds in your head.
2. If you get to 10 and all layers are loaded, you should not see any warning alerts saying they were slow. If layers happen to take longer, it becomes unscientific since you mostly cannot tell what time was counted as loading and what time as drawing. Code looks good tho eh!
3. Switch to Sample 34. This sample has the global default set to half a second.
4. Watch layers load, pan and zoom map around. Unless your connection is a real rocket, should be getting plenty of warning alerts in the :bell: bucket.


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ramp4-pcar4/ramp4-pcar4/2251)
